simple_network_sim/network_of_populations/__init__.py

```diff
@@ -536,61 +536,69 @@ def getInternalInfectiousContactsInNode(
+                    contacts_series,
+                    infectious_series,
+                    susceptibles,
+                    totalInAge,
+                    random_state,
+                )
+            else:
+                infectiousContacts[ageTo] = acc
 
     return infectiousContacts
 
 
 def _computeInfectiousContactsStochastic(
-        contacts: float,
-        infectious: float,
-        susceptibles: float,
-        totalInAge: float,
-        random_state: np.random.Generator,
+        contacts: List[float],
+        infectious: List[float],
+        susceptibles: Union[int, float],
+        totalInAge: Union[int, float],
+        random_state: np.random.Generator
 ) -> float:
     """
-    From raw contacts (between any two people in different age groups), filters only those contacts that originated from
-    an infectious person and received by a susceptible person. The contact did not yet lead to a new infection, we need
-    to multiply by the infection probability.
-
-    :param contacts: The number of reported contacts
-    :param infectious: The number of infectious people from which the contacts
-                       can originate
-    :param susceptibles: The number of susceptible people to which the contacts
-                         can target
+    Takes a list of contacts (usually coming from other age groups) which should be paired with a list of infectious
+    people from the same group. This function will then estimate the number of infectious contacts those people will
+    produce.
+
+    :param contacts: A list with the number of contacts per incoming group
+    :param infectious: The number of incoming infectious people, paired with the the list of contacts
+    :param susceptibles: The number of susceptible people to which the contacts can target
     :param totalInAge: Total number of people in target population
     :param random_state: Random number generator used for the model
-    :return: The number of potentially infectious contacts.
+    :return: The number of potentially infectious contacts
     """
-    assert isinstance(infectious, int) or infectious.is_integer()
+    if len(contacts) != len(infectious):
+        raise ValueError("contacts and infectious must have the same length")
     assert isinstance(totalInAge, int) or totalInAge.is_integer()
     assert isinstance(susceptibles, int) or susceptibles.is_integer()
 
-    numberOfContacts = random_state.poisson(contacts, size=int(infectious))
-    # If 0, the hypergeometric calls fails, but mathematically is 0
-    numberOfContacts = numberOfContacts[numberOfContacts > 0]
+    inf_max = int(round(max(infectious)))
+    # each np.array represent one original (contact, infectious) pair
+    numbersOfContacts: List[np.array] = []
+    for contact, infected in zip(contacts, infectious):
+        x = random_state.poisson(contact, int(round(infected)))
+        numbersOfContacts.append(np.concatenate([x, np.zeros(inf_max - len(x), dtype=int)]))
```

Comment:
What do you think about initializing a `np.zeros((inf_max, len(infection)))`, and then assigning into it row by row, rather than the more tedious list of arrays creation?

Reply:
Sounds good.